.github/workflows/infra_apply.yaml

```
@@ -132,7 +159,7 @@ jobs:
   tf_apply:
     name: 'Terraform Apply'
     runs-on: ${{ inputs.use_labels && inputs.use_private_agent && inputs.environment || inputs.use_private_agent && 'self-hosted' || 'ubuntu-latest' }}
-    needs: [tf_plan]
+    needs: ["get-terraform-version", "tf-modules-check", "tf_plan"]
```

Perhaps we can exclude modules check?
```suggestion
    needs: [get-terraform-version, tf_plan]
```